File: 5주차/이지혜/BinaryTree.java
import javax.sound.midi.Soundbank;

public class BinaryTree {
    Node rootNode = null;


    public static void main(String[] args) {
        int count = 7; //정점의 개수
        Node[] nodeList = new Node[count+1]; // index의 시작이 0부터라서

        for(int i=1; i <= count; i++){
            Node binaryTree = new Node(i);
            nodeList[i] = binaryTree;
        }

        for(int i=1; i <= count;i++) {
            if( i*2 <= count) {
                nodeList[i].left = nodeList[i *2];
                nodeList[i].right = nodeList[(i*2)+1];
            }
        }

        preOrder(nodeList[1]);
    }
    static void preOrder(Node node){
        if(node != null) {
            System.out.println(node.value + "  ");
            preOrder(node.left);
            preOrder(node.right);
        }
    }

    static void inOrder(Node node){
        if(node != null){
            inOrder(node.left);
            System.out.println(node.value + "  ");
            inOrder(node.right);
        }
    }

    static void postOder(Node node){
        if(node != null){
            postOder(node.left);
            postOder(node.right);
            System.out.println(node.value+"  ");
       }
    }


}

File: 5주차/이영훈/binary_tree.py
class Node:
    def __init__(self, val=None):
        self.val = val
        self.left = None
        self.right = None

class BinaryTree:
    def __init__(self):
        self.root = None

    def preorder(self, node):
        if node != None:
            print(node.val, end=' ')
            if node.left:
                self.preorder(node.left)
            if node.right:
                self.preorder(node.right)

    def inorder(self, node):
        if node != None:
            if node.left:
                self.inorder(node.left)
            print(node.val, end=' ')
            if node.right:
                self.inorder(node.right)

    def postorder(self, node):
        if node != None:
            if node.left:
                self.postorder(node.left)
            if node.right:
                self.postorder(node.right)
            print(node.val, end=' ')

    # def levelorder(self, node):

if __name__ == "__main__":
    tree = BinaryTree()
    n1 = Node(10)
    n2 = Node(20)
    n3 = Node(30)
    n4 = Node(40)
    n5 = Node(50)
    n6 = Node(60)
    n7 = Node(70)
    n8 = Node(80)

    tree.root = n1
    n1.left = n2
    n1.right = n3
    n2.left = n4
    n2.right = n5
    n3.left = n6
    n3.right = n7
    n4.left = n8

    tree.preorder(tree.root)
    print()

    tree.inorder(tree.root)
    print()

    tree.postorder(tree.root)